pkg/run/install_dashboard_test.go

```diff
@@ -38,3 +45,191 @@ var _ = Describe("InstallDashboard", func() {
 		Expect(err.Error()).To(Equal(applyAllErrorMsg))
 	})
 })
+
+var _ = Describe("GenerateManifestsForDashboard", func() {
+	var fakeLogger *loggerfakes.FakeLogger
+
+	BeforeEach(func() {
+		fakeLogger = &loggerfakes.FakeLogger{}
+	})
+
+	It("generates manifests successfully", func() {
+		valuesMap := map[string]interface{}{
+			"adminUser": map[string]interface{}{
+				"create":       true,
+				"passwordHash": "test-secret",
+				"username":     "admin",
+			},
+		}
+
+		helmRepository := &sourcev1.HelmRepository{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       sourcev1.HelmRepositoryKind,
+				APIVersion: sourcev1.GroupVersion.Identifier(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ww-gitops",
+				Namespace: "test-namespace",
+			},
+			Spec: sourcev1.HelmRepositorySpec{
+				URL: "https://helm.gitops.weave.works",
+				Interval: metav1.Duration{
+					Duration: time.Minute,
+				},
+			},
+		}
+
+		values, err := json.Marshal(valuesMap)
+		Expect(err).NotTo(HaveOccurred())
+
+		helmRelease := &helmv2.HelmRelease{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       helmv2.HelmReleaseKind,
+				APIVersion: helmv2.GroupVersion.Identifier(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ww-gitops",
+				Namespace: "test-namespace",
+			},
+			Spec: helmv2.HelmReleaseSpec{
+				Interval: metav1.Duration{
+					Duration: time.Minute,
+				},
+				Chart: helmv2.HelmChartTemplate{
+					Spec: helmv2.HelmChartTemplateSpec{
+						Chart:   "weave-gitops",
+						Version: "2.0.6",
+						SourceRef: helmv2.CrossNamespaceObjectReference{
+							Kind: "HelmRepository",
+							Name: "ww-gitops",
+						},
+						ReconcileStrategy: "ChartVersion",
+					},
+				},
+				Values: &v1.JSON{Raw: values},
+			},
+		}
+
+		expectedHelmRepository, err := yaml.Marshal(helmRepository)
+		Expect(err).NotTo(HaveOccurred())
+
+		expectedHelmRelease, err := yaml.Marshal(helmRelease)
+		Expect(err).NotTo(HaveOccurred())
+
+		divider := []byte("---\n")
+
+		expected := append(divider, expectedHelmRepository...)
+		expected = append(expected, divider...)
+		expected = append(expected, expectedHelmRelease...)
+
+		actual, err := run.GenerateManifestsForDashboard(fakeLogger, secret, helmRepository, helmRelease)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(actual).To(Equal(expected))
+	})
+})
+
+var _ = Describe("MakeHelmRelease", func() {
+	var fakeLogger *loggerfakes.FakeLogger
+
+	BeforeEach(func() {
+		fakeLogger = &loggerfakes.FakeLogger{}
+	})
+
+	It("creates helmrelease successfully", func() {
+		valuesMap := map[string]interface{}{
+			"adminUser": map[string]interface{}{
+				"create":       true,
+				"passwordHash": "test-secret",
+				"username":     "admin",
+			},
+		}
+
+		values, err := json.Marshal(valuesMap)
+		Expect(err).NotTo(HaveOccurred())
+
+		helmRelease := &helmv2.HelmRelease{
```

Comment:
These tests would probably require less maintenance if they were turned "inside out", so rather than serialising the expected data and comparing it exactly, we could de-serialize the generated data and comparing "some" of it to what we expect. I think that would make it lighter to add more tests so we could test e.g. namespace and other flags, but wouldn't have to change it every time the chart version changed.

Reply:
Thanks, good idea, will try it!